way_point[] way_points